warn-stealers.lua

```diff
@@ -0,0 +1,107 @@
+-- Script to warn when creatures that may steal food become visible
+--[====[
+warn-stealers
+=============
+Will make a zoomable announcement whenever a creature that can eat food, guzzle drinks, or steal items enters the map and moves into a revealed location.
+Takes ``start`` or ``stop`` as parameters.
+]====]
+
+local persistTable = require("persist-table")
+local eventful = require("plugins.eventful")
+local repeatUtil = require("repeat-util")
+
+local eventfulKey = "warn-stealers"
+local numTicksBetweenChecks = 100
+
+if df.global.gamemode ~= df.game_mode.DWARF then
+    if df.global.gamemode ~= df.game_mode.NONE then
+        -- errors when gamemode is NONE
+        persistTable.GlobalTable.warnStealersCache = nil
+    end
+    return
+end
+
+if not persistTable.GlobalTable.warnStealersCache then
+    persistTable.GlobalTable.warnStealersCache = {}
+end
+local cache = persistTable.GlobalTable.warnStealersCache
+
+local function isUnitHidden(unit)
+    local block = dfhack.maps.getTileBlock(unit.pos)
+    if not block then
+        return false
+    end
+    return block.designation[unit.pos.x%16][unit.pos.y%16].hidden
+end
+
+local races = df.global.world.raws.creatures.all
+
+local function addToCacheIfStealerAndHidden(unitId)
+    local unit = df.unit.find(unitId)
```

Comment:
do we need to check the game mode here and stop if we're not where we expect to be?